rpi-source

```diff
@@ -408,8 +438,13 @@ if args.default_config or not kernel.config:
     info(".config (generating default)")
     if processor_type == 0:
         sh("cd %s && make bcmrpi_defconfig" % (linux_symlink,))
```

Comment:
I think it's better to put an explicit `ARCH=%s` with the value of `architecture_type` in each of these `make` lines, which should also remove the need for the trailing comma.

Reply:
In fact, with so much in common, isn't it better to use processor_type and possibly architecture_type to generate a defconfig_name, and use that in a single `sh()` line?

Reply:
That's probably a good idea.  Perhaps, we should do that in a separate PR that optimizes the way this is all handled by `rpi-source`. What do you think?

Reply:
Something like this:
```patch
 if args.default_config or not kernel.config:
     info(".config (generating default)")
+    defconfig = None
     if processor_type == 0:
-        sh("cd %s && make bcmrpi_defconfig" % (linux_symlink,))
+        defconfig = "bcmrpi_defconfig"
     elif processor_type == 1:
-        sh("cd %s && make bcm2709_defconfig" % (linux_symlink,))
+        defconfig = "bcm2709_defconfig"
     elif processor_type == 2:
         if architecture_type == 0:
-            sh("cd %s && make bcm2709_defconfig" % (linux_symlink,))
+            defconfig = "bcm2709_defconfig"
         elif architecture_type == 1:
-            sh("cd %s && make bcm2711_defconfig" % (linux_symlink,))
+            defconfig = "bcm2711_defconfig"
     elif processor_type == 3:
-        sh("cd %s && make bcm2711_defconfig" % (linux_symlink,))
+        defconfig = "bcm2711_defconfig"
     elif processor_type == 4:
         if page_size == 4096:
-            sh("cd %s && make bcm2711_defconfig" % (linux_symlink,))
+            defconfig = "bcm2711_defconfig"
         elif page_size == 16384:
-            sh("cd %s && make bcm2712_defconfig" % (linux_symlink,))
+            defconfig = "bcm2712_defconfig"
     else:
         fail("Unsupported processor_type %d" % processor_type)
+    if defconfig is not None:
+        sh("cd %s && make %s" % (linux_symlink, defconfig))
 else:
     info(".config")
     writef(os.path.join(linux_dir, '.config'), kernel.config)
```

Reply:
Resolved in [bd2423c](https://github.com/RPi-Distro/rpi-source/pull/29/commits/bd2423c00790fc875b9df3e9b452456451834db2)